front/components/data_source/RequestDataSourcesModal.tsx

```diff
@@ -0,0 +1,200 @@
+                      CONNECTOR_CONFIGURATIONS[
+                        selectedDataSourceIntegration.connectorProvider
+                      ].name
+                    }
+                    icon={
+                      CONNECTOR_CONFIGURATIONS[
+                        selectedDataSourceIntegration.connectorProvider
+                      ].logoComponent
+                    }
+                  />
+                ) : (
+                  <Button
+                    label="Pick your platform"
+                    variant="tertiary"
+                    size="sm"
+                    type="select"
+                  />
+                )}
+              </DropdownMenu.Button>
+              <DropdownMenu.Items width={180}>
+                {filteredDataSourceIntegrations.map((ds) => (
+                  <DropdownMenu.Item
+                    key={ds.dataSourceName}
+                    label={ds.name}
+                    onClick={() => setSelectedDataSourceIntegration(ds)}
+                    icon={
+                      CONNECTOR_CONFIGURATIONS[ds.connectorProvider]
+                        .logoComponent
+                    }
+                  />
+                ))}
+              </DropdownMenu.Items>
+            </DropdownMenu>
+          )}
+        </div>
+
+        {selectedDataSourceIntegration && (
+          <div>
+            <p className="s-mb-2 s-text-sm s-text-element-700">
+              The administrator for{" "}
+              {
+                CONNECTOR_CONFIGURATIONS[
+                  selectedDataSourceIntegration.connectorProvider
+                ].name
+              }{" "}
+              is {selectedDataSourceIntegration.editedByUser?.fullName}. Send an
+              email to {selectedDataSourceIntegration.editedByUser?.fullName},
+              explaining your request.
+            </p>
+            <TextArea
+              placeholder={`Hello ${selectedDataSourceIntegration.editedByUser?.fullName},`}
+              value={message}
+              onChange={setMessage}
+              className="s-mb-2"
+            />
+            <Button
+              label="Send"
+              variant="primary"
+              size="sm"
+              onClick={async () => {
+                const userEmail =
+                  selectedDataSourceIntegration?.editedByUser?.email;
+                if (!userEmail || !selectedDataSourceIntegration) {
+                  sendNotification({
+                    type: "error",
+                    title: "Error sending email",
+                    description:
+                      "An unexpected error occurred while sending email.",
+                  });
+                } else {
+                  try {
+                    await sendRequestDataSourceEmail({
+                      email: userEmail,
+                      emailMessage: message,
+                      dataSourceName: selectedDataSourceIntegration.name,
+                      emailRequester: currentUserEmail,
```

Comment:
IMHO we should infer this in the endpoint using the `auth`.

Reply:
ok